Apps/phsentinelone/sentinelone_connector.py

```diff
@@ -646,6 +646,67 @@ def _handle_get_threat_info(self, param):
         action_result.add_data(response)
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _handle_get_applications(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        ip_hostname = param.get('ip_hostname')
+        params = None
+        if ip_hostname:
+            try:
+                ret_val = self._get_computer_name(ip_hostname, action_result)
+            except Exception:
+                return action_result.set_status(phantom.APP_ERROR, "Did not get proper response from the server")
+            self.save_progress('Agent query: {}'.format(ret_val))
+
+            if ret_val == '0':
+                return action_result.set_status(phantom.APP_ERROR, "Endpoint not found")
+            elif ret_val == '99':
+                return action_result.set_status(phantom.APP_ERROR, "More than one endpoint found")
+            else:
+                summary = action_result.update_summary({})
+                summary['ip_hostname'] = ip_hostname
+                summary['computer_name'] = ret_val
+                params = {"agentComputerName__contains": ret_val}
+        header = self.HEADER
+        header["Authorization"] = "APIToken %s" % self.token
+        ret_val, response = self._make_rest_call('/web/api/v2.1/installed-applications', action_result, headers=header, params=params)
+        action_result.add_data(response)
+        self.save_progress("Ret_val: {0}".format(ret_val))
+        if phantom.is_fail(ret_val):
+            self.save_progress("Failed to get applications.  Error: {0}".format(action_result.get_message()))
+            return action_result.get_status()
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_get_cves(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        ip_hostname = param.get('ip_hostname')
+        params = None
+        if ip_hostname:
+            try:
+                ret_val = self._get_computer_name(ip_hostname, action_result)
+            except Exception:
+                return action_result.set_status(phantom.APP_ERROR, "Did not get proper response from the server")
+            self.save_progress('Agent query: {}'.format(ret_val))
+
+            if ret_val == '0':
+                return action_result.set_status(phantom.APP_ERROR, "Endpoint not found")
+            elif ret_val == '99':
+                return action_result.set_status(phantom.APP_ERROR, "More than one endpoint found")
+            else:
+                summary = action_result.update_summary({})
+                summary['ip_hostname'] = ip_hostname
+                summary['computer_name'] = ret_val
```

Comment:
populate the param in case user specifies the hostname.
check get_applications function.